### PR TITLE
PHOENIX-7063 Track and account garbage collected phoenix connections

### PR DESCRIPTION
**Pull request by lokiore**
FYI @jpisaac

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java [hunk 1]
```diff
@@ -69,12 +69,14 @@ public ConnectionQueryServicesTestImpl(QueryServices services, ConnectionInfo in
     
     @Override
     public void addConnection(PhoenixConnection connection) throws SQLException {
+        super.addConnection(connection);
```
Comment (jpisaac) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java:72]: Was this added intentionally? Do see a similar statement below?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java [hunk 2]
```diff
@@ -69,12 +69,14 @@ public ConnectionQueryServicesTestImpl(QueryServices services, ConnectionInfo in
...
         connections.add(connection);
         super.addConnection(connection);
     }
     
     @Override
     public void removeConnection(PhoenixConnection connection) throws SQLException {
+        super.removeConnection(connection);
```
Comment (jpisaac) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java:79]: Same here

**Review by jpisaac — COMMENTED** [review]
LGTM, except for some additional extraneous stmts

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java (hunk 1 above)
Comment (lokiore) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java:72]: Yes, We are testing connection counts for different scenarios in LoggingConnectionLimiterIT, and these counts gets updated when we call acquire connection in CQSI's addConnection, which was not being used before as part of CQSTestImpl.

Comment (jpisaac) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java:72]: But there is already a super.addConnection(..) below?

Comment (lokiore) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ConnectionQueryServicesTestImpl.java:72]: Aaah I see....... ✌️

**Review by jpisaac — APPROVED** [review]
LGTM +1

**Pull request merged**